NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -91,132 +182,135 @@ def run(
+            E = np.sqrt(S_per_bin / (scipy.constants.c * units.m / units.s * scipy.constants.epsilon_0 * (
+                        units.coulomb / units.V / units.m))) / d_f
+
+            # assign random phases to electric field
+            noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=np.complex128)
+            phases = np.random.uniform(0, 2. * np.pi, len(S))
+
+            noise_spectrum[1][passband_filter] = np.exp(1j * phases) * E
+            noise_spectrum[2][passband_filter] = np.exp(1j * phases) * E
+
+            channel_noise_spec = np.zeros_like(noise_spectrum)
+
+            for channel in station.iter_channels():
                 if detector.get_relative_position(station.get_id(), channel.get_id())[2] < 0:
-                    t_theta = geometryUtilities.get_fresnel_t_p(zenith, n_ice, 1)
-                    t_phi = geometryUtilities.get_fresnel_t_s(zenith, n_ice, 1)
-                    fresnel_zenith = geometryUtilities.get_fresnel_angle(zenith, n_ice, 1.)
-                    if fresnel_zenith is None:
-                        continue
+                    curr_t_theta = t_theta
+                    curr_t_phi = t_phi
+                    curr_fresnel_zenith = fresnel_zenith
                 else:
-                    t_theta = 1
-                    t_phi = 1
-                    fresnel_zenith = zenith
+                    curr_t_theta = 1
+                    curr_t_phi = 1
+                    curr_fresnel_zenith = zenith
+
+                antenna_pattern = self.__antenna_pattern_provider.load_antenna_pattern(
+                    detector.get_antenna_model(station.get_id(), channel.get_id()))
+                antenna_orientation = detector.get_antenna_orientation(station.get_id(), channel.get_id())
+
+                # calculate the phase offset in comparison to station center
+                # consider additional distance in air & ice
+                # assume for air & ice constant index of refraction
+                channel_pos_x, channel_pos_y = detector.get_relative_position(station.get_id(), channel.get_id())[
+                    [0, 1]]
+                channel_depth = abs(min(detector.get_relative_position(station.get_id(), channel.get_id())[2], 0))
+                sin_zenith = np.sin(zenith)
+                delta_phases = (
+                        2 * np.pi * freqs[passband_filter] / (scipy.constants.c * units.m / units.s) * n_air *
+                        (
+                                sin_zenith *
+                                (np.cos(azimuth) * channel_pos_x + channel_pos_y * np.sin(azimuth)) +
+                                channel_depth *
+                                ((n_ice / n_air) ** 2 + sin_zenith ** 2) /
+                                np.sqrt((n_ice / n_air) ** 2 - sin_zenith ** 2)
+                        )
+                )
+
+                # add random polarizations and phase to electric field
+                polarizations = np.random.uniform(0, 2. * np.pi, len(S))
```

Comment:
...also an open question whether random polarizations are right (technically, this implies that the signal from each signal is fully linearly polarized in a random direction for each antenna. I don't think this is right, we should think about how to do this correctly, even if the impact in the end is probably small).

Reply:
Hm but this is because we use the same phase in the projection of the two polarisations right?